exporters/trace/fluentforward/fluentforward.go

```diff
@@ -0,0 +1,199 @@
+package fluentforward
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net"
+	"time"
+
+	apitrace "go.opentelemetry.io/otel/api/trace"
+	export "go.opentelemetry.io/otel/sdk/export/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/label"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+type ffSpan struct{
+	_msgpack struct{}            `msgpack:",asArray"`
+	Tag string                   `msgpack:"tag"`
+	Ts int64                     `msgpack:"ts"`
+	SpanData spanData						 `msgpack:"spanData"`
+}
+
+type spanData struct{
+	SpanContext spanContext         `msgpack:"spanContext"`
+	ParentSpanId string             `msgpack:"parentSpanId"`
+	SpanKind apitrace.SpanKind      `msgpack:"spanKind"`
+	Name string                     `msgpack:"name"`
+	StartTime int64                 `msgpack:"startTime"`
+	StatusMessage string            `msgpack:"statusMessage"`
+	EndTime int64                   `msgpack:"endTime"`
+	Attrs map[label.Key]interface{} `msgpack:"attrs"`
+	DroppedAttributeCount int       `msgpack:"droppedAttributesCount"`
+	DroppedMessageEventCount int    `msgpack:"droppedMessageEventCount"`
+	DroppedLinkCount int            `msgpack:"droppedLinkCount"`
+	StatusCode string               `msgpack:"statusCode"`
+	Links []link                    `msgpack:"links"`
+}
+
+type spanContext struct{
+	TraceId string  `msgpack:"traceId"`
+	SpanId string   `msgpack:"spanId"`
+	TraceFlags byte `msgpack:"TraceFlags"`
+}
+
+type link struct{
+	SpanContext spanContext         `msgpack:"spanContext"`
+	Attrs map[label.Key]interface{} `msgpack:"attrs"`
+}
+
+type event struct{
+	Attrs map[label.Key]interface{} `msgpack:"attrs"`
+	DroppedAttributeCount int       `msgpack:"droppedAttributesCount"`
+}
+
+type Exporter struct{
+	url string
+	serviceName string
+	client *net.TCPConn
+	o options
+}
+
+type Option func(*options)
+
+type options struct{
+	config *sdktrace.Config
+	logger *log.Logger
+}
+
+func WithLogger(logger *log.Logger) Option{
+	return func(o *options)	{
+		o.logger = logger
+	}
+}
+
+func WithSDK(config *sdktrace.Config) Option{
+	return func(o *options){
+		o.config = config
+	}
+}
+
+func InstallNewPipeline(ffurl, serviceName string, opts ...Option) error{
+	tp,err := NewExportPipeline(ffurl, serviceName, opts...)
+	if err!=nil{
+		return err
+	}
+	global.SetTracerProvider(tp)
+	return nil
+}
+
+func NewExportPipeline(ffurl, serviceName string, opts ...Option) (*sdktrace.TracerProvider, error){
```

Comment:
is this method have to be public? This qq may be answered by updating the README.md

Reply:
readme should explain when public methods have to be used